src/systems/filecoin_mining/storage_mining/_index.md

```diff
@@ -27,3 +27,11 @@ Filecoin Storage Mining Subsystem
 # Sector in StorageMiner State Machine (both)
 
 {{< diagram src="diagrams/sector_fsm.dot.svg" title="Sector State Machine (both)" >}}
+
+# ElectionPoSt
+
+To enable short PoSt response time, miners are required submit a PoSt when they are elected to mine a block, hence PoSt on election or ElectionPoSt. When miners win a block, they need to immediately generate a PoStProof and submit that along with the ElectionProof. Both the ElectionProof and PoStProof are checked at Block Validation by `StoragePowerConsenusSubsystem`. When a block is included, a special message is added that calls `SubmitElectionPoSt` which will process sector updates in the same way as successful `SubmitSurprisePoSt` do. Note that it will be a single PoStProof in a single `SubmitElectionPoSt` message instead of a Commit-then-Prove scheme.
+
+# SurprisePoSt
```

Comment:
I want to make the point that we actually don't need the full surprise post and can cut on the blockchain's memory footprint:

**Context**: One can see SurprisePost as two inner components: the "surprise" component and the "deterministic" component. The "surprise" part is what we understand (randomly sample at every block). The "deterministic" part is when a miner haven't been challenged in a given period (`MIN_CHALLENGE_PERIOD`) then that miner *has* to submit its post.

**Election Post**: In that setting, the way I understand it, *every* miner has to check whether he is eligible for producing a block or not. In a sense, that is the *same* behavior as surprise post already: every miner has to check at every block if they have to submit their post! The only difference is that in ElectionPost this check is private, and in SurprisePost this check is public. The point here is that in both cases, miner **are already on the hook every round**. They have to perform this check every round. 

Hence, I believe we **don't need the surprise component** of SurprisePost, but only the deterministic one. In other words, I would recommend to only spec the fact that if a miner hasn't submitted its post (therefore a block) after `MIN_CHALLENGE_PERIOD`), it has to submit its post, that's it.

Reply:
@nikkolasg yes, for security of power sure, but for the miner's storage & state on chain, no -- we need to clean up the state in the chain automatically, and we need to deal with cases where miner gets faulted but needs to recover. Look at the actor code, sector state machine, etc

Reply:
What you telling about here sounds (1) more about engineering issues rather than security issues (clean up state?) or (2) issues that surprise post is not meant to solve at all (recover faults?). 
- What I am proposing if anything should simplify things rather than adding complexity. 
- We should address engineering issues explicitely and not as a by-product of another (security) feature, as this is a dangerous road that leads to more interconnected components without clear boundaries and with fuzzy links (which I believe we're trying to avoid)